### DISCUSSION TITLE
Docker non-interactive container producing ERD diagram from a given sqlite file

### DISCUSSION THREAD

**Discussion by r-scotti** [schemacrawler/SchemaCrawler#596] [Q&A] [answered]
Hello. 
Testing different solutions to produce the ERD schema diagram from a give sqlite file, SchemaCrawler pops up for the quality of the output.
My objective is to produce the diagram 'non-interactively'.
Based on the instructions available at https://github.com/schemacrawler/SchemaCrawler-Docker#run-the-schemacrawler-command-line, with fome corrections i manage to see the process completed through the following steps:
launch 
`sudo docker run -v $(pwd):/share --rm -i -t --entrypoint=/bin/bash schemacrawler/schemacrawler`
then, in container bash-5.1$ : 
`schemacrawler --server=sqlite --database=/share/chinook-database-2.0.1.sqlite --info-level=minimum --command=schema --output-file=/share/schema.png`
I transferred the commands to a Dockerfile to build an image that can be used non-interactively
`# syntax=docker/dockerfile:1`
`FROM schemacrawler/schemacrawler`
`ENTRYPOINT /bin/bash`
`VOLUME /share`
`ENV PATH=$PATH:/opt/schemacrawler/lib`
`RUN /opt/schemacrawler/bin/schemacrawler.sh --server=sqlite --database=sc.db --info-level=minimum --command=schema --output-file=/share/schema.png`
but the build
`DOCKER_BUILDKIT=1 # https://docs.docker.com/engine/reference/builder/#entrypoint`
`sudo docker build -t nfs/sc .`
 aborts with following main message:
> SchemaCrawler 16.16.10
> Error: Graphviz was not available to create the requested diagram.
> Please reinstall Graphviz, and make it available on the system PATH. If you are unable to install Graphviz, make https://github.com/nidi3/graphviz-java jar files available to SchemaCrawler, in the lib/ folder.
> ...
> Environment:  SchemaCrawler 16.16.10,   Linux 5.13.0-27-generic,  Oracle Corporation OpenJDK 64-Bit Server VM 17-ea+14,   SQLite 3.36.0,   SQLite JDBC 3.36.0.3

Complete session log is here: [error.html.pdf](https://github.com/schemacrawler/SchemaCrawler/files/7995201/error.html.pdf)
As far as I can see (which is not very far at all!) Graphviz is available on the system PATH:
`bash-5.1$ ls /opt/schemacrawler/lib/graphviz-java-0.18.1.jar
/opt/schemacrawler/lib/graphviz-java-0.18.1.jar`
I added the path to the PATH system variable to be sure.
I am stuck at this point.
What am I missing? Is my strategy not good?
Many thanks for any help, Roberto

**Comment by sualeh** [comment]
@r-scotti - the https://github.com/schemacrawler/SchemaCrawler-Docker project is archived, so any information on that project is outdated.

**Reply by r-scotti** [reply]
Hello, thanks for answering.
I am aware that the project is archived but I could not find a better support: is there a new project that covers the topic 'schemacrawler in a Docker container'?
I have never dealt with Docker or containers before so I found the information in the GitHub pages very useful as starting point.
With some corrections I could get the flow to complete in the interactive session.
I got stuck trying to build the image that should proceed without interaction.
I have to understand why schemacrawler, in the build session, does not find GraphViz.
Can somebody help me?
Many thanks,
Roberto

**Comment by sualeh** [comment, marked as answer]
@r-scotti - please follow the instructions on [How to Visualize Your SQLite Database with One Command (and Nothing to Install)](https://dev.to/sualeh/how-to-visualize-your-sqlite-database-with-one-command-and-nothing-to-install-1f4m)

**Reply by r-scotti** [reply]
Many many thanks! perfetto! Roberto
PS - Nothing to install beside Docker and the pull of SchemaCrawler image, but this is easy.

**Reply by sualeh** [reply]
Glad you got it working, Roberto @r-scotti
